src/training/vqa_fine_tune.py

```diff
@@ -0,0 +1,316 @@
+import argparse
+
+import pandas as pd
+import torch
+from PIL import Image
+from torch import nn
+from torch.utils.data import Dataset, DataLoader
+import torch.nn.functional as F
+from tqdm import tqdm
+
+import open_clip
+from open_clip.factory import get_tokenizer
+from training.scheduler import cosine_lr
+from training.train import AverageMeter
+import evaluate
+from sklearn import preprocessing
+import numpy as np
+
+from datasets import load_dataset_builder
+from datasets import load_dataset
+
+class VQATextDataset(Dataset):
+    def __init__(self, df, split, transforms, labelencoder, tokenizer=None):
+        self.df = df
+        self.transforms = transforms
+        self.tokenize = tokenizer
+        self.labels = labelencoder.transform(df['multiple_choice_answer'])
+    def __len__(self):
+        return len(self.df)
+
+    def __getitem__(self, idx):
+        item = self.df.iloc[idx]
+        img_path = item["image"]["path"]
+        image = Image.open(str(img_path))
+        text = item["question"]
+        label = self.labels[idx]
+        return {
+            'image': self.transforms(image),
+            'text': self.tokenize([text])[0],
```

Comment:
You would need the attention mask from this no? Or is it automatically calculated on the fly?